### PR TITLE
docs: add pseudo states for calendar sticker sheet

### PR DESCRIPTION
**Pull request by NatalieLoyzaga**
## Why

Add pseudo states for Calendar to be sent to Chromatic

## What

- Use storybook .play() to get elements and add className focus-visible and hover story class.
- Checked chromatic to ensure the popovers aren't overlapping

image

### CODE REVIEW

File: packages/date-picker/package.json [hunk 1]
```diff
@@ -43,7 +43,8 @@
   },
   "devDependencies": {
     "@kaizen/design-tokens": "^9.0.3",
-    "@types/classnames": "^2.3.1"
+    "@types/classnames": "^2.3.1",
+    "@storybook/testing-library": "^0.0.13"
```
Comment (HeartSquared) [inline packages/date-picker/package.json:47]: I'd put this in the root `package.json`

File: packages/date-picker/src/DatePicker/components/Calendar/Calendar.scss [hunk 2]
```diff
@@ -251,3 +251,18 @@ $rdp-cell-size: 40px;
   background-color: $color-blue-100;
   color: $color-blue-500;
 }
+
+// Add Storybook psuedo classes
```
Comment (HeartSquared) [inline packages/date-picker/src/DatePicker/components/Calendar/Calendar.scss:255]:
```suggestion
// Add Storybook pseudo classes
```

File: packages/date-picker/src/DatePicker/components/Calendar/Calendar.scss [hunk 3]
```diff
@@ -251,3 +251,18 @@ $rdp-cell-size: 40px;
...
+:global(.story__day-hover) {
```
Comment (HeartSquared) [inline packages/date-picker/src/DatePicker/components/Calendar/Calendar.scss:256]: Ideally, we'd be able to put this class just wherever we've got the styles already defined. This way, we wouldn't need to change multiple places to ensure they align.

File: packages/date-picker/src/DatePicker/components/Calendar/Calendar.scss [hunk 4]
```diff
@@ -251,3 +251,18 @@ $rdp-cell-size: 40px;
...
+  &:not([aria-disabled="true"]) {
+    &:not(.outside) {
```
Comment (HeartSquared) [inline packages/date-picker/src/DatePicker/components/Calendar/Calendar.scss:258]: We should probably revisit this `.outside` class and find out exactly what it's doing.

**Review by HeartSquared — COMMENTED** [review]
Leaving a few comments first. Haven't looked at the stories yet.

File: packages/date-picker/src/DatePicker/components/Calendar/Calendar.scss (hunk 2 above)
Comment (NatalieLoyzaga) [reply packages/date-picker/src/DatePicker/components/Calendar/Calendar.scss:255]: i hate this word ahahh

File: packages/date-picker/docs/Calendar.stories.tsx [hunk 5]
```diff
@@ -43,24 +44,82 @@ const CalendarExample = (
...
+    { calendar: "default", name: "5th September (Monday)" },
+    { calendar: "selected", name: "5th September (Monday)" },
+    { calendar: "navigation", name: "Go to previous month" },
+  ]
+
+  calendars.forEach(cal => {
+    getElementWithinCalendar(`${cal.calendar}-hover`, cal.name).classList.add(
+      "story_hover"
+    )
+    getElementWithinCalendar(`${cal.calendar}-focus`, cal.name).classList.add(
```
Comment (HeartSquared) [inline packages/date-picker/docs/Calendar.stories.tsx:118]:
```suggestion
  calendars.forEach(({ calendar, name }) => {
    getElementWithinCalendar(`${calendar}-hover`, name).classList.add(
      "story_hover"
    )
    getElementWithinCalendar(`${calendar}-focus`, name).classList.add(
```

File: packages/date-picker/docs/Calendar.stories.tsx [hunk 6]
```diff
@@ -43,24 +44,82 @@ const CalendarExample = (
...
+
+  const getElementWithinCalendar = (id: string, name: string) => {
+    const calendar = canvas.getByTestId(id)
+    return within(calendar).getByRole("button", {
+      name,
+    })
+  }
+
+  const calendars = [
+    { calendar: "default", name: "5th September (Monday)" },
```
Comment (HeartSquared) [inline packages/date-picker/docs/Calendar.stories.tsx:109]:
```suggestion
    { row: "default", name: "5th September (Monday)" },
```

Moreso a nitpick, but I got a little confused thinking `calendar` was referencing a single Calendar, but it seems to be referencing a row of calendars

File: packages/date-picker/docs/Calendar.stories.tsx [hunk 7]
```diff
@@ -43,24 +44,82 @@ const CalendarExample = (
...
+      </StoryWrapper.Row>
+    </StoryWrapper>
+  </div>
 )
 
 export const StickerSheetCalendar = StickerSheetCalendarTemplate.bind({})
 StickerSheetCalendar.storyName = "Sticker Sheet (Default)"
 StickerSheetCalendar.parameters = {
-  chromatic: { disable: false },
+  chromatic: { disable: false, viewports: [1800] },
```
Comment (HeartSquared) [inline packages/date-picker/docs/Calendar.stories.tsx:94]: No longer needed since the popper isn't there to cause weirdness anymore :)

File: packages/date-picker/docs/Calendar.stories.tsx [hunk 8]
```diff
@@ -43,24 +44,82 @@ const CalendarExample = (
 const StickerSheetCalendarTemplate: Story<{ isReversed: boolean }> = ({
   isReversed,
 }) => (
-  <StoryWrapper isReversed={isReversed}>
-    <StoryWrapper.RowHeader headings={["Selected Date", "Disabled Dates"]} />
-    <StoryWrapper.Row rowTitle="Calendar">
-      <CalendarExample
-        id="calendar-dialog-selected"
-        value={new Date(2022, 1, 5)}
+  <div style={{ height: "1600px" }}>
```
Comment (HeartSquared) [inline packages/date-picker/docs/Calendar.stories.tsx:49]: No longer needed since the popper isn't there to cause weirdness anymore :)

File: packages/date-picker/src/_primitives/Calendar/Calendar.scss [hunk 9]
```diff
@@ -98,7 +98,8 @@ $rdp-cell-size: 40px;
     border: $border-solid-border-width $border-solid-border-style transparent;
     border-radius: $cell-border-radius;
 
-    &:hover {
+    &:hover,
+    &:global(.story_hover) {
```
Comment (HeartSquared) [inline packages/date-picker/src/_primitives/Calendar/Calendar.scss:102]: Since it's a global class we should be more specific ([bem is helpful here](http://getbem.com/naming/)), otherwise we might end up with global styles from different component sticker sheets affecting others.

```suggestion
    &:global(.story__datepicker__calendar--hover) {
```

File: packages/date-picker/docs/Calendar.stories.tsx [hunk 10]
```diff
@@ -30,37 +31,97 @@ export default {
...
 ): JSX.Element => (
-  <Calendar
-    mode="single"
-    weekStartsOn={DayOfWeek.Sun}
-    onDayChange={() => undefined}
-    defaultMonth={new Date(2022, 1, 5)}
-    locale={enAU}
-    {...props}
-  />
+  <div data-testid={props.id}>
```
Comment (HeartSquared) [inline packages/date-picker/docs/Calendar.stories.tsx:34]: I was wondering why we couldn't just put this on `Calendar` itself (it's what I would expect) and it turns out we don't accept further props.

While we could certainly put it in now, the question is whether it should go on the container or the DayPicker. 
Ideally I'd expect to extend DayPicker because it's the main element, but it appears we can't actually remove the container because DayPicker doesn't accept `ref`, `id`, `role`, of which case it means `data-testid` should therefore go on the container.

Strange predicament here, but I suppose as a subcomponent it can be less flexible 🤷🏻 

TLDR; you can either leave it, or add `...restProps` to Calendar and put it in the container div (remember to use `OverrideClassName<HTMLAttributes<HTMLDivElement>>` if doing so).

**Review by HeartSquared — APPROVED** [review]
Love your work! 💯
